api/src/main/java/org/apache/cloudstack/acl/APIChecker.java

```diff
@@ -27,4 +28,6 @@
     // If false, apiChecker is unable to handle the operation or not implemented
     // On exception, checkAccess failed don't allow
     boolean checkAccess(User user, String apiCommandName) throws PermissionDeniedException;
+
+    boolean checkAccess(Account account, User user, String apiCommandName, Role role) throws PermissionDeniedException;
```

Comment:
```suggestion
    boolean checkAccess(Account account, User user, String apiCommandName, Role role, Boolean isDynamicApiCheckerEnabled) throws PermissionDeniedException;
```

Reply:
@shwstppr 
as said, I suggest to add `isDynamicApiCheckerEnabled` in `checkAccess` method.

Reply:
@weizhouapache made some changes to query only enabled APICheckers for checkAccess.
Also, realized my earlier tests were a bit skewed as I should be comparing vanilla 4.16 branch with this PR build instead of only trim parameter usage of the API. Will test and report them
cc @rohityadavcloud